packages/stream/StreamClient.ts

```diff
@@ -680,6 +618,107 @@ export default class StreamClient {
+   */
+  private async prepareStreamTransaction(
+    recipient: MultiRecipient,
+    streamParams: CreateMultiParams
+  ) {
+    const {
+      sender,
+      mint,
+      start,
+      period,
+      cliff,
+      canTopup,
+      cancelableBySender,
+      cancelableByRecipient,
+      transferableBySender,
+      transferableByRecipient,
+      automaticWithdrawal = false,
+      withdrawalFrequency = 0,
+      partner = null,
+    } = streamParams;
+    let ixs: TransactionInstruction[] = [];
+    const commitment =
+      typeof this.commitment == "string"
+        ? this.commitment
+        : this.commitment.commitment;
+    const recipientPublicKey = new PublicKey(recipient.recipient);
+    const mintPublicKey = new PublicKey(mint);
+    const metadata = Keypair.generate();
+    const [escrowTokens] = await web3.PublicKey.findProgramAddress(
+      [Buffer.from("strm"), metadata.publicKey.toBuffer()],
+      this.programId
+    );
+
+    const senderTokens = await ata(mintPublicKey, sender.publicKey);
+    const recipientTokens = await ata(mintPublicKey, recipientPublicKey);
+    const streamflowTreasuryTokens = await ata(
+      mintPublicKey,
+      STREAMFLOW_TREASURY_PUBLIC_KEY
+    );
+
+    const partnerPublicKey = partner
+      ? new PublicKey(partner)
+      : STREAMFLOW_TREASURY_PUBLIC_KEY;
+
+    const partnerTokens = await ata(mintPublicKey, partnerPublicKey);
+
+    ixs.push(
+      createStreamInstruction(
+        {
+          start: new BN(start),
+          depositedAmount: recipient.depositedAmount,
+          period: new BN(period),
+          amountPerPeriod: recipient.amountPerPeriod,
+          cliff: new BN(cliff),
+          cliffAmount: recipient.cliffAmount,
+          cancelableBySender,
+          cancelableByRecipient,
+          automaticWithdrawal,
+          transferableBySender,
+          transferableByRecipient,
+          canTopup,
+          name: recipient.name,
+          withdrawFrequency: new BN(
+            automaticWithdrawal ? withdrawalFrequency : period
+          ),
+        },
+        this.programId,
+        {
+          sender: sender.publicKey,
+          senderTokens,
+          recipient: new PublicKey(recipient.recipient),
+          metadata: metadata.publicKey,
+          escrowTokens,
+          recipientTokens,
+          streamflowTreasury: STREAMFLOW_TREASURY_PUBLIC_KEY,
+          streamflowTreasuryTokens: streamflowTreasuryTokens,
+          partner: partnerPublicKey,
+          partnerTokens: partnerTokens,
+          mint: new PublicKey(mint),
+          feeOracle: FEE_ORACLE_PUBLIC_KEY,
+          rent: SYSVAR_RENT_PUBKEY,
+          timelockProgram: this.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+          withdrawor: WITHDRAWOR_PUBLIC_KEY,
+          systemProgram: SystemProgram.programId,
+        }
+      )
+    );
+    let hash = await this.connection.getRecentBlockhash(commitment);
```

Comment:
having it here means it will be fetched for each stream, maybe one per chunk is better (less calls, faster everything is done)

Reply:
I will try that and test if it will not raise any "recent blockhash not found" error

Reply:
@nemanjab17 since I use recent blockhash as a key for batch items I am going to keep fetching blockhash on creation. Can I Resolve this issue?